File: DSA_Level-1/Basics Of Programming/2D Arrays/Saddle Price/SaddlePrice_Solution.java

import java.io.*;
import java.util.*;

public class Main {

  public static void main(String[] args) throws Exception {
    Scanner scn = new Scanner(System.in);
    int n = scn.nextInt();
    int[][] nums = new int[n][n];

    for (int i = 0; i < n; ++i) {
      for (int j = 0; j < n; ++j) {
        nums[i][j] = scn.nextInt();
      }
    }

    for (int i = 0; i < nums.length; ++i) {
      int min = nums[i][0];
      int col = 0;
      for (int j = 0; j < nums[i].length; ++j) {
        if (min > nums[i][j]) {
          min = nums[i][j];
          col = j;
        }
      }

      int k = 0;
      while ( k < nums.length && nums[k][col] <= min) {
        ++k;
      }

      if (k == nums.length) {
        System.out.println(min);
        return;
      }

    }

    System.out.println("Invalid input");
  }

}

File: DSA_Level-1/Basics Of Programming/2D Arrays/Saddle Price/SaddlePrice_Solution.js

function saddlePrice(nums){
    for(let i=0;i<nums.length;++i){
        let min=nums[i][0];
        let col=0;
        for(let j=0;j<nums[i].length;++j){
            if(min>nums[i][j]){
                min=nums[i][j];
                col=j;
            }
        }

        let k=0;
        while( k<nums.length && nums[k][col]<=min){
            ++k;
        }

        if(k==nums.length){
            return min;
        }

    }

    return "Invalid Input";
}

console.log(saddlePrice([[11,12,13,14],[21,22,23,24],[31,32,33,34],[41,42,43,44]]));